tests/providers/test_ssn.py

```diff
@@ -630,8 +630,8 @@ def setUp(self):
         Faker.seed(0)
 
 
-class TestEsMX(unittest.TestCase):
-    def setUp(self):
+class TestEsMX:
+    def setup_class(self):
```

Comment:
the reason why we are using `unittest.TestCase.setUp` is that we want to reset the `.seed` before each test. Can this be left as it is?

Reply:
Hi @fcurella! I changed it to a pytest-style class because I wanted to use `pytest.mark.parametrize` to test the parameter options exhaustively in a slightly cleaner way.

It seems there are some options that would allow us to preserve that:

1) Using [`setup_method`](https://docs.pytest.org/en/7.4.x/how-to/xunit_setup.html#method-and-function-level-setup-teardown)
2) Using a pytest fixture. The default scope is `function`

I confirm that at least the latter works, so wdyt?

And I can certainly leave `setUp` as is and rewrite the parametrized test if you still prefer that approach.